### PR TITLE
Fix reducing very big caches

### PR DESCRIPTION
**Pull request by janekolszak**
Script that copies the cache neglects problematic batches, described in #8

### CODE REVIEW

File: tools/rewrite.ts [hunk 1]
```diff
@@ -33,27 +37,60 @@ async function main() {
...
+  const chunkSize = 1000
+
+  const endChunk = Math.min(Math.ceil(numKeys / chunkSize), args.startChunk + args.numChunks)
+
+  let counter = 0
+  let i
+  for (i = args.startChunk; i < endChunk; i++) {
+    try {
+      output.transactionSync(() => {
+        // input.getKeys({ snapshot: false, limit: chunkSize, offset: i * chunkSize })
```
Comment (ppedziwiatr) [inline tools/rewrite.ts:58]: if that's no longer needed - please remove

File: tools/rewrite.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
```
Comment (ppedziwiatr) [inline tools/rewrite.sh:1]: would it be possible to make this script runnable via npx?

File: tools/rewrite.sh [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        exit 1
+    fi
+}
+
+run() {
+    export NODE_OPTIONS="-r ts-node/register --no-warnings --max-old-space-size=4000"
+    for i in $(seq 0 $NUM_CHUNKS_IN_BATCH 100); do
+        node ./rewrite.ts -i $INPUT -o $OUTPUT -s $i -c $NUM_CHUNKS_IN_BATCH
+        result=$?
+        if [ $result = 12 ]; then
```
Comment (ppedziwiatr) [inline tools/rewrite.sh:70]: hmm..what exactly is `12` here?

Comment (ppedziwiatr) [reply tools/rewrite.sh:70]: oh..I see in the rewrite.ts script...

File: tools/rewrite.ts [hunk 4]
```diff
@@ -33,27 +37,60 @@ async function main() {
...
+      })
+
+      await output.flushed
+
+    } catch (err) {
+      console.error("Failed to chunk, exception:", i, err)
+      process.exit(1)
+    }
+  }
+  i -= 1
```
Comment (ppedziwiatr) [inline tools/rewrite.ts:83]: why is the `i` decreased here?

File: tools/rewrite.ts [hunk 5]
```diff
@@ -33,27 +37,60 @@ async function main() {
...
+  console.log("Inserted elements ", i * chunkSize + counter, "of", numKeys, "(", Math.floor((100 * i * chunkSize + counter) / numKeys), "%)")
+  if (counter !== 0) {
```
Comment (ppedziwiatr) [inline tools/rewrite.ts:85]: I don't understand how it works :-( the counter is only increased, but never decreased - so it always will be != 0?

File: tools/rewrite.sh (hunk 3 above)
Comment (janekolszak) [reply tools/rewrite.sh:70]: Just a hard coded value. It's random, I rolled the dice honestly...

**Pull request merged**